katdal/datasources.py

```diff
@@ -168,37 +167,25 @@ def view_capture_stream(telstate, capture_block_id=None, stream_name=None):
     """
     # Detect the capture block
     if not capture_block_id:
-        capture_blocks = []
-        if 'sdp_capture_block_id' in telstate:
-            for value_time in telstate.get_range('sdp_capture_block_id', st=0):
-                cbid = str(value_time[0])
-                if 'obs_params' in telstate.view(cbid, exclusive=True):
-                    capture_blocks.append(cbid)
-        if not capture_blocks:
+        try:
+            capture_block_id = str(telstate['capture_block_id'])
```

Comment:
For my education, why do we have both sdp_capture_block_id and capture_block_id? Can we unify them?

Reply:
capture_block_id and the stream_name and stream_type are inserted by the meta writer just before writing, since at this point we are totally sure about the CBID and stream for this dump. I need to go and check where sdp_capture_block_id comes from and how it choses which CBID to use.

Reply:
sdp_capture_block_id is a mutable key written by master controller as it inits each capture block - it's used by the session to find the current CBID for example.

Would it be sensible to use that same name for the key injected by the metadata writer?

Reply:
My reasoning for choosing a non sdp key is exactly to distinguish from these kinds of internal things - the idea being this is definitive statement (sdp or not) about the data in this RDB represents - so basically a system property.

Reply:
I'm still far from convinced, but let's bikeshed it face-to-face.

Reply:
Remember to invite me too :-)

Reply:
So far Simon has beaten me into leaving it as is, with the possibility to later rename sdp_capture_block_id to capture_block_id. The reason for not doing it now is that the metadata writer has no way to replace keys, only to add new ones, and it would require some refactoring to support that.

Reply:
I think if you simply add the self-referential `<cbid>_capture_block_id` (yes, another one!) all will be well and whole. The `sdp_capture_block_id` is very useful to find all CBs in a multi-CB telstate (how else?). The `<cbid>_capture_block_id` is meant to be "the" capture block ID in a viewed single-CB telstate. Then @sratcliffe's improper `capture_block_id` is a similar construct for an unviewed single-CB telstate, which is styled to match `<cbid>_capture_block_id` after applying its view.

I'm not in favour of proper telstate keys at the top level that do not start with an identifiable subsystem / component, e.g. `sdp_`, so I would not like to see `sdp_capture_block_id` turn into `capture_block_id`.

Reply:
Also, another avenue I mentioned to @sratcliffe is that we could "preview" all keys in the relevant <stream>, <cbid> and <cbid_stream> views by stripping off their prefixes and promoting them to the top level in the RDB files. Then you could open the file and and you would automatically be in the correct view. This also resolves the discrepancy between `capture_block_id` and `sdp_capture_block_id` (the former is a flattened version of the self-referential `<cbid>_capture_block_id` and therefore not a standalone key).